### PR TITLE
[GraphQL/Validators][EASY] Clean-up convert_to_validators

### PR DESCRIPTION
**Pull request by amnn**
## Description

Refactoring `convert_to_validators` to:

- Avoid using chains and nests of `and_then`s or `map`s (these are hard to follow).
- Avoid cloning the system state multiple times to extract out parts of it.

## Test Plan

```
sui-graphql-rpc$ cargo nextest run
```

## Stack
- #15467 
- #15470 
- #15471 
- #15472 
- #15473
- #15474 
- #15475 
- #15484 
- #15485
- #15519
- #15520

### CODE REVIEW

**Review by wlmyng — APPROVED** [review]
this is nice, thanks!

File: crates/sui-graphql-rpc/src/context_data/db_data_provider.rs [hunk 1]
```diff
@@ -1805,31 +1805,36 @@ pub(crate) fn convert_to_validators(
...
+        validator_report_records,
+        ..
+    }) = system_state
+    {
+        (
+            BTreeMap::from_iter(at_risk_validators),
+            BTreeMap::from_iter(validator_report_records),
+        )
+    } else {
+        Default::default()
```
Comment (stefan-mysten) [inline crates/sui-graphql-rpc/src/context_data/db_data_provider.rs:1819]: Not sure about this default. Could you explain the rationale behind returning default instead of None? Does default return None?

**Review by stefan-mysten — APPROVED** [review]

File: crates/sui-graphql-rpc/src/context_data/db_data_provider.rs (hunk 1 above)
Comment (wlmyng) [reply crates/sui-graphql-rpc/src/context_data/db_data_provider.rs:1819]: It would create a tuple of empty BTreeMaps, instead of (None, None)

easier to work with empty BTreeMap than Option<BTreeMap>, latter would require more unwraps or Some checks

Comment (stefan-mysten) [reply crates/sui-graphql-rpc/src/context_data/db_data_provider.rs:1819]: I see, so it infers that the return type is BTreeMap so calling default will return an empty BTreeMap, got it. Thank you!
